packages/nextjs/src/index.client.ts

```diff
@@ -50,6 +49,19 @@ export function init(options: NextjsOptions): void {
     return;
   }
 
+  let tunnelPath: string | undefined;
+  if (globalWithInjectedValues.__sentryRewritesTunnelPath__ && options.dsn) {
+    const dsnComponents = dsnFromString(options.dsn);
+    if (dsnComponents.host.match(/^o\d+\.ingest\.sentry\.io$/)) {
+      const orgId = dsnComponents.host.split('.')[0];
+      tunnelPath = `${globalWithInjectedValues.__sentryRewritesTunnelPath__}/${orgId}/${dsnComponents.projectId}/`;
+      __DEBUG_BUILD__ && logger.info(`Tunneling events to "${tunnelPath}"`);
+      options.tunnel = tunnelPath;
+    } else {
+      __DEBUG_BUILD__ && logger.warn('Provided DSN is not a Sentry SaaS DSN. Will not tunnel events.');
```

Comment:
l: This should work with self-hosted as well right? Why indicate `SaaS`?

Reply:
Is there a reason to tunnel if you're already on self-hosted? Requests to self-hosted instances are probably not being blocked by ad blockers.

Another reason we limit this is because of next.js rewrite restrictions. For example, the protocol needs to be static.

Reply:
I was going to ask the same thing. It's for sure less likely for someone's self-hosted instance to land on a block-list, but not impossible.

Reply:
> Is there a reason to tunnel if you're already on self-hosted

Good point, not sure actually, don't know enough to comment on this. @kamilogorek is tunneling a thing for self-hosted users?

Reply:
> It's for sure less likely for someone's self-hosted instance to land on a block-list, but not impossible.

I believe if somebody's self-hosted instance is landing on a block-list, they're probably high-profile enough that their proxy route (via this option) will also land on a block list - rendering the option useless anyhow.

I personally would ship this without directly supporting self-hosted. And if we see demand for proxying to other hosts we can pretty easily add something like a `tunnelTarget` option that would allow for that. It should be perfectly compatible with this change.

I am not gonna go into detail here but there are a few things to consider when letting users choose the host. For example, we would need some special logic to tunnel requests to `ingest.sentry.io` because you cannot put that string into a tunnel request, since it would be blocked.

Reply:
They can also implement this logic themselves if they need to for the self-hosted case. Yeah that sounds reasonable enough for me.

Reply:
> I believe if somebody's self-hosted instance is landing on a block-list, they're probably high-profile enough that their proxy route (via this option) will also land on a block list - rendering the option useless anyhow.
> 
> I personally would ship this without directly supporting self-hosted. And if we see demand for proxying to other hosts we can pretty easily add something like a `tunnelTarget` option that would allow for that. It should be perfectly compatible with this change.

Yup, that's fair. Let's leave it as is.